tasks/workflow/worker.go

```diff
@@ -602,6 +606,7 @@ func (w *worker) runUndeploy(ctx context.Context, t *taskExecution) {
 			log.Printf("%+v", err)
 			w.checkAndSetDeploymentStatus(ctx, t.targetID, deployments.UNDEPLOYMENT_FAILED)
 			checkAndSetTaskStatus(ctx, t.kv, t.taskID, t.step, tasks.TaskStatusFAILED)
+			events.PublishAndLogWorkflowStatusChange(ctx, w.consulClient.KV(), t.targetID, t.taskID, "uninstall", tasks.TaskStatusFAILED.String())
```

Comment:
During an undeployment, seeing many times the event `Deployment:  <deployment ID> Deployment Status: undeployment_in_progress`.

Maybe we could change line 603 above, that is calling for now:
```
w.setDeploymentStatus(ctx, t.targetID, deployments.UNDEPLOYMENT_IN_PROGRESS)
```
It could be changed to set the deployment status to `undeployment_in_progress` only if the undeployment isn't already in progress. But we should not replace the call `w.setDeploymentStatus()` by the current implementation of `w.checkAndSetDeploymentStatus()` because this function implementation would refuse to undeploy a failed deployment.

Reply:
Not sure to understand

Reply:
each time a new execution is started, an event `Deployment Status: undeployment_in_progress` is generated, so we have many of such events, while we should have just one event `Deployment Status: undeployment_in_progress` generated.

It works fine for the deployment part: one and only one event `Deployment Status: deployment_in_progress` is generated.

Reply:
I just pushed a check as you mention